### ISSUE TITLE
Let Controller run at AJAX page change

### ISSUE THREAD

**Issue by mpsijm** [thervh70/ContextProject_RDD#32]
GitHub does not load a new page when clicking on a link, but rather refreshes the current DOM-tree via an AJAX request. The extension should be able to log actions happening in this new DOM-tree.

[Chrome Extension Documentation: run_at](https://developer.chrome.com/extensions/content_scripts#run_at)
Following from this, we will need to create an event handler which triggers on DOM refresh.

**Comment by mpsijm** [comment]
The AJAX request works, but there is a similar bug that causes dynamically created DOM elements to not be logged, because at the moment the Controller only hooks to DOM on URL change.

**Comment by MathiasMeuleman** [comment]
Should be fixed when implementing the right getCallBack() methods in the respective classes.

**Comment by mpsijm** [comment]
*tl;dr:* Takes too long, not enough time to fix this during this sprint XD
-
Okay, I have done some research and try-outs, and would like to document this for others when trying to fix this bug. I have tried:
1. In the constructor of ClickEventElementBinding, change:
```diff
     constructor(private elementSelectionBehaviour: ElementSelectionBehaviour) {
-        const elements = elementSelectionBehaviour.getElements();
+        this.setEventHandler();
+    }
+
+    private setEventHandler() {
+        const elements = this.elementSelectionBehaviour.getElements();
         elements.off(this.eventType);
         elements.on(
             this.eventType,
-            this.elementSelectionBehaviour.getCallback(this.eventID)
+            function() {
+                self.elementSelectionBehaviour.getCallback(this.eventID);
+                self.setEventHandler();
+            }
         );
     }
```
This will cause only the already existing elements to be re-hooked. This means, when clicking e.g. the inline-comment-button, only the inline-comment-buttons will be re-hooked while the cancel-inline-comment button will not be hooked.

2. I passed the ContentController to the `ElementSelectionBehaviour`s and let the callbacks call `ContentController.hookToDOM()` (after making that function public), essentially refreshing the entire DOM tree on every hover/click/keypress/everything. This made the extension incredibly slow.

What _should_ be done when we have more time, is re-building this part of the architecture in a way that makes the DOM refreshed only on (relevant) clicks. This could be done with an observer pattern or rewriting how the `ESB`s and `ElementEventBinding`s interact.
